### DISCUSSION TITLE
Example of how to use a SQL Server database with EF Core

### DISCUSSION THREAD

**Discussion by facundoLecuna** [testcontainers/testcontainers-dotnet#834] [Q&A]
I have an API REST .NET 7 and I want to build integrations tests using a sql server container database. I've tried to follow the example in the documentation (https://dotnet.testcontainers.org/examples/aspnet/), but it seems to be outdated for version 3.0.0, some classes are missing or obsolete.
What I want is to load the sql server, create a database and then be able to work with the Entity Framework Core DbSets.
Is there an example for the new version? 
Thanks

**Comment by HofmeisterAn** [comment]
Thank you for bringing this to our attention. You are correct, that particular example is no longer accurate. I will rewrite it within the next few days. In the meantime, you can refer to the complete example [here](https://github.com/testcontainers/testcontainers-dotnet/tree/develop/examples/WeatherForecast/tests), which already uses version 3.0.0. Alternatively, you can look into [this guide](https://github.com/dotnet/AspNetCore.Docs/pull/28531/files), which is based on Microsoft's article titled [Integration tests in ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-7.0). We will soon publish this as a dedicated guide.

**Comment by sandeep-eroad** [comment]
I am also facing issue with MsSqlBuilder, I have set it up as shown here

``` c#
private readonly MsSqlContainer dbContainer = new MsSqlBuilder()
        .WithImage("mcr.microsoft.com/mssql/server:2022-latest")
        .WithCleanUp(true)
        .WithWaitStrategy(Wait.ForUnixContainer().UntilPortIsAvailable(1433))
        .Build();
```

and then in `InitializeAsync` I do the migration by calling
``` c#
await dbContext.Database.MigrateAsync();
```

I can see table `__EFMigrationsHistory` in `master` db but no other tables are created and there is no exception raised.

Not sure if it's me or if others have also seen this issue, need some help here.

**Reply by HofmeisterAn** [reply]
> I am also facing issue with MsSqlBuilder, I have set it up as shown here

Testcontainers modules are pre-configured. There is usually no reason to override the default configuration. Your configuration overrides the default configuration and indicates readiness of the container way too early. Please use the default configuration:

```csharp
_ = new MsSqlBuilder().Build()
```
